core/lib/Thelia/Core/Template/Loop/Image.php

```diff
@@ -361,11 +361,19 @@ public function parseResults(LoopResult $loopResult)
                 // Dispatch image processing event
                 $this->dispatcher->dispatch(TheliaEvents::IMAGE_PROCESS, $event);
 
+                $originalImageSize = getimagesize($sourceFilePath);
```

Comment:
I think you should use `$event->getOriginalFileUrl()` instead of `$sourceFilePath` here.